lib/src/component/http_req.rs

```diff
@@ -396,32 +398,26 @@ impl http_req::Host for Session {
         h: http_types::RequestHandle,
         b: http_types::BodyHandle,
         backend_name: String,
-    ) -> Result<
-        // This return type is a little surprising. Please see the [error-detail] note at the top
-        // of the file for an explanation.
-        Result<http_types::Response, (Option<http_req::SendErrorDetail>, types::Error)>,
-        anyhow::Error,
-    > {
+    ) -> Result<http_types::Response, (Option<http_req::SendErrorDetail>, types::Error)> {
         // This initial implementation ignores the error detail field
-        match self.send(h, b, backend_name).await {
-            Ok(r) => Ok(Ok(r)),
-            Err(e) => e.with_empty_detail(),
-        }
+        self.send(h, b, backend_name)
+            .await
+            .map_err(types::Error::with_empty_detail)
```

Comment:
If all errors here are going to return `(None, e)`, why are we including the `None`? Is this a future expansion thing?

Reply:
Yep. The wiggle implementation always sets the error-detail to a default value, but in the wit version, I opted to represent that as an `option<send-error-detail>` instead. Returning `None` is the same as the wiggle version, but I didn't want to break our ability to return the additional detail later.